packages/cli/src/lib/services/config/schema.ts

```diff
@@ -55,6 +55,30 @@ export const OptionsHooksSchema = z.object({
   snapshot: SnapshotHooksSchema.optional(),
 } satisfies { [key in z.infer<typeof CollectionEnum> | 'snapshot']: z.Schema });
 
+const ClientConfigSchema = z.object({
+  globals: z
+    .object({
+      URL: z.instanceof(URL).optional(),
+      WebSocket: z.function().optional(),
+      fetch: z.function().returns(z.instanceof(Promise)).optional(),
+      logger: z
+        .object({
+          log: z.function(),
+          info: z.function(),
+          warn: z.function(),
+          error: z.function(),
+        })
+        .optional(),
+    })
+    .optional(),
+});
+
+const RestConfigSchema = z.object({
+  credentials: z.string().optional(),
+  onRequest: z.function().optional(),
+  onResponse: z.function().optional(),
+});
+
```

Comment:
You could add `passthrough` to these schemas so the user can pass additional options that may be accepted in future versions of Directus but are not expected by Directus-Sync. Since Directus-Sync does not use the contents of these `ClientConfig` and `RestConfig` objects, there will be no typing issues. This will simply make the runtime more flexible.